index.js

```diff
@@ -446,13 +466,28 @@ instance.prototype.init_actions = function (system) {
 					default: 'self',
 					choices: self.CHOICES_SURFACES,
 				},
+				{
+					type: 'textwithvariables',
```

Comment:
Yeah, I agree here. My thinking was that extra input makes more sense than whole new action.

However, if there's some sort of new combo-input field I think that could work. It seems like using variables in workflows is very popular, so trying to  allow that flexibility while keeping things clean I think is important.

Reply:
There isnt anything yet, Ive been putting off making the ui components for it.

Reply:
dropdowns do have the 'allowCustom' option which does this, but im not sure if that is appropriate here..
I would like to see it as probably a checkbox before/after each input field, which toggles them into a 'variable parsing' mode, or perhaps that mode should be a dropdown listing every known variable?. With companion-core parsing the string before handing it off to the module. And probably an opt-in property for each input field that modules have to opt into as it will probably cause various crashes if the wrong input type is passed in

In my head there is a fairly thought out plan, its a case of finding motivation to fiddle with ui stuff to get it to work